src/common/file_posix.c

```diff
@@ -216,73 +214,5 @@ util_ddax_region_find(const char *path)
 {
 	LOG(3, "path \"%s\"", path);
 
-	int dax_reg_id_fd;
-	char dax_region_path[PATH_MAX];
-	char reg_id[DAX_REGION_ID_LEN];
-	char *end_addr;
-	os_stat_t st;
-
-	ASSERTne(path, NULL);
-	if (os_stat(path, &st) < 0) {
-		ERR("!stat \"%s\"", path);
-		return -1;
-	}
-
-	dev_t dev_id = st.st_rdev;
-
-	unsigned major = os_major(dev_id);
-	unsigned minor = os_minor(dev_id);
-	int ret = util_snprintf(dax_region_path, PATH_MAX,
-		"/sys/dev/char/%u:%u/device/dax_region/id",
-		major, minor);
-	if (ret < 0) {
-		ERR("!snprintf");
-		return -1;
-	}
-
-	if ((dax_reg_id_fd = os_open(dax_region_path, O_RDONLY)) < 0) {
-		LOG(1, "!open(\"%s\", O_RDONLY)", dax_region_path);
-		return -1;
-	}
-
-	ssize_t len = read(dax_reg_id_fd, reg_id, DAX_REGION_ID_LEN);
-
-	if (len == -1) {
-		ERR("!read(%d, %p, %d)", dax_reg_id_fd,
-			reg_id, DAX_REGION_ID_LEN);
-		goto err;
-	} else if (len < 2 || reg_id[len - 1] != '\n') {
-		errno = EINVAL;
-		ERR("!read(%d, %p, %d) invalid format", dax_reg_id_fd,
-			reg_id, DAX_REGION_ID_LEN);
-		goto err;
-	}
-
-	int olderrno = errno;
-	errno = 0;
-	long reg_num = strtol(reg_id, &end_addr, 10);
-	if ((errno == ERANGE && (reg_num == LONG_MAX || reg_num == LONG_MIN)) ||
-			(errno != 0 && reg_num == 0)) {
-		ERR("!strtol(%p, %p, 10)", reg_id, end_addr);
-		goto err;
-	}
-	errno = olderrno;
-
-	if (end_addr == reg_id) {
-		ERR("!strtol(%p, %p, 10) no digits were found",
-			reg_id, end_addr);
-		goto err;
-	}
-	if (*end_addr != '\n') {
-		ERR("!strtol(%s, %s, 10) invalid format",
-			reg_id, end_addr);
-		goto err;
-	}
-
-	os_close(dax_reg_id_fd);
-	return (int)reg_num;
-
-err:
-	os_close(dax_reg_id_fd);
-	return -1;
+	return  pmem2_device_dax_region_find(path);
```

Comment:
Shouldn't pmem2_device_dax_region_find accept fds instead of paths? I don't know why you need this function in pmem2 though, so I might be wrong.

Reply:
or even os_stat_t instead of fd/path

Reply:
or even os_stat_t instead of fd/path